hazelcast/src/test/java/com/hazelcast/map/impl/tx/MapTransactionTest.java

```diff
@@ -408,14 +413,19 @@ public Boolean execute(TransactionalTaskContext context) throws TransactionExcep
         });
         assertTrue(b);
 
-        IMap map1 = h2.getMap("putWithTTL");
-        assertEquals("value", map1.get("1"));
+        final Object result = map2.get("1");
+        if (deletedCounter.get() > 0) {
+            System.out.println("Value was already removed from map, quitting test");
```

Comment:
In the test case we are expecting the value to be evicted eventually, so we don't need to test against value insertion but value eviction.

So we can refactor the remaining test to an `assertTrueEventually` block which checks the `deletedCounter.get()>0` and `assertNull(map2.get("1"))`. This way we can also get rid of the `sleep`